app/controllers/discovered_hosts_controller.rb

```diff
@@ -73,6 +73,7 @@ def update
   def perform_update(host, success_message = nil)
     Taxonomy.no_taxonomy_scope do
       ::ForemanDiscovery::HostConverter.set_build_clean_facts(host)
+      @host.primary_interface.discovery_queue_rebuild_dns unless @host.primary_interface.pending_dns_record_changes?
```

Comment:
This code will not work with auto-provisioning, make sure it work in both ways. Anyways.

I think instead of calling this explicitly, we should create new `after_validation :queue_dns_discovery` hook. In the hook method, check if we are converting from Discovered, like we do in other hooks, then queue it when necessary.

Add tests for normal and autoprovisioning please. And make the commit message shorter :-)